### PR TITLE
Remove old mocha dependency and outdated ruby/rails

### PR DESCRIPTION
**Pull request by pschambacher**
/cc @zendesk/classic-upgrade

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,20 +8,16 @@ branches:
   only: master
 
 rvm:
-  - 2.1.8
-  - 2.2.4
-  - 2.3.0
+  - 2.2.7
+  - 2.3.4
```
Comment (craig-day) [inline .travis.yml]: add `2.4.x` too?

Comment (pschambacher) [inline .travis.yml]: It explodes, will review later

**Comment by craig-day** [comment]
:+1:

**Comment by grosser** [comment]
should also declare the gem `>= ruby 2.2` to be safe ...

**Pull request merged**